jmx-metrics/src/main/groovy/io/opentelemetry/contrib/jmxmetrics/GroovyMetricEnvironment.java

```diff
@@ -33,15 +39,48 @@
   private final SdkMeterProvider meterProvider;
   private final Meter meter;
 
-  // Observer updaters can only be specified in the builder as of v0.13.0, so to work with our model
+  private class Registry<T> {
+    private final Map<Integer, AtomicReference<T>> registry = new ConcurrentHashMap<>();
+
+    private AtomicReference<T> getExisting(
+        String name,
+        String description,
+        String unit,
+        InstrumentType instrumentType,
+        InstrumentValueType instrumentValueType) {
+      InstrumentDescriptor descriptor =
+          InstrumentDescriptor.create(name, description, unit, instrumentType, instrumentValueType);
+      registry.putIfAbsent(descriptor.hashCode(), new AtomicReference<>());
```

Comment:
Should be able to return [`computeIfAbsent`](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ConcurrentHashMap.html#computeIfAbsent-K-java.util.function.Function-) rather than `putIfAbsent` & then return `get`.